Documentation/automatic-bootstrap-artifact.md

```diff
@@ -0,0 +1,70 @@
+Automatic bootstrap Artifact
+============================
+
+The bootstrap Artifact provides a way to populate the Mender client database when it is first
+initialized, since it is difficult to update the binary database from the build. Among other things,
+this enables delta updates to be performed using the very first update, instead of needing to do one
+rootfs update first to populate the `artifact_provides` attributes.
+
+Properties of the bootstrap Artifact
+------------------------------------
+
+A bootstrap Artifact is an empty Artifact which provides only meta-data. Specifically, it contains
+`artifact_provides` key/value pairs.
+
+To be considered a bootstrap Artifact, the Artifact must comply with the following requirements:
+* Have no `scripts`
+* Have no `artifact_depends`
+* Contain exactly 1 update which:
+  * Contains no files
+  * Include `artifact_provides`
+
+The bootstrap Artifact generated via [`meta-mender`](https://github.com/mendersoftware/meta-mender)
+or [`mender-convert`](https://github.com/mendersoftware/mender-convert) contain
+`rootfs-image.checksum` and `rootfs-image.version` keys, enabling therefore delta Artifacts after
+installation of. However custom implementations could have other `artifact_provides` keys and it
+will still be considered valid.
+
+
+Automatic installation of the Artifact
+--------------------------------------
+    
+On start-up, Mender checks for the existence of an the bootstrap Artifact in path
+`/var/lib/mender/bootstrap.mender` and installs it in order to initialize the device database.
+
+This applies both for `daemon` command and `bootstrap` and `install` standalone calls.
+    
+The Artifact is not installed if the device already has a database. When the Artifact is not found
+(and the database is empty) the database is initialized with `artifact-name=unknown`.
```

Comment:
Also upon errors, installing it, right? :thinking:

Reply:
Yes, good catch! Upon errors it will set it to unknown and delete the file -> I will add that.